sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -62,11 +67,36 @@ export class Sender {
   /**
    * Closes the underlying AMQP sender link.
    * Once closed, the sender cannot be used for any further operations.
-   * Use the `createSender` function on the QueueClient or TopicClient to instantiate a new Sender
+   * Use the `createSender` function on the EventHubClient to instantiate a new Sender
    *
    * @returns {Promise<void>}
    */
   async close(): Promise<void> {
-    this._isClosed = true;
+    try {
+      if (this._context.connection && this._context.connection.isOpen() && this._eventHubSender) {
+        await this._context.senders[this._eventHubSender.name].close();
```

Comment:
It is possible that the `this._context.senders` map will not have an entry for this name. So check if it exists before calling `close()` on it